Apromore-Custom-Plugins/Log-Importer-Logic/src/main/java/org/apromore/service/logimporter/services/legacy/AbstractLogImporter.java

```diff
@@ -0,0 +1,183 @@
+        for (Map.Entry<String, String> entry : eventAttributes.entrySet()) {
+            if (entry.getValue() != null && entry.getValue().trim().length() != 0) {
+                attribute = new XAttributeLiteralImpl(entry.getKey(), entry.getValue());
+                xEvent.getAttributes().put(entry.getKey(), attribute);
+            }
+        }
+        return xEvent;
+    }
+
+    protected void assignEventsToTrace(LogEventModel logEventModel, XTrace xTrace) {
+        XEvent xEvent;
+
+        if (logEventModel.getStartTimestamp() != null) {
+            xEvent = createEvent(logEventModel, false);
+            xTrace.add(xEvent);
+        }
+        xEvent = createEvent(logEventModel, true);
+        xTrace.add(xEvent);
+    }
+
+    protected void assignMyCaseAttributes(Map<String, String> caseAttributes, XTrace xTrace) {
+        XAttributeMap xAttributeMap = xTrace.getAttributes();
+
+        if (caseAttributes != null && !caseAttributes.isEmpty()) {
+            XAttribute attribute;
+            for (Map.Entry<String, String> entry : caseAttributes.entrySet()) {
+                if (entry.getValue() != null && entry.getValue().trim().length() != 0
+                    && !xAttributeMap.containsKey(entry.getKey())) {
+                    attribute = new XAttributeLiteralImpl(entry.getKey(), entry.getValue());
+                    xTrace.getAttributes().put(entry.getKey(), attribute);
+                }
+            }
+        }
+    }
+
+    protected boolean isValidLineCount(int lineCount) {
+        return config == null || config.getMaxEventCount() == null || lineCount <= config.getMaxEventCount();
+    }
+
+    protected void constructTrace(
+        final TreeMap<String, XTrace> tracesHistory, final LogEventModel logEventModel,
+        final XFactory xFactory, XConceptExtension concept) {
+        if (tracesHistory.isEmpty() || !tracesHistory.containsKey(logEventModel.getCaseID())) {
+            XTrace xT = xFactory.createTrace();
+            concept.assignName(xT, logEventModel.getCaseID());
+            assignEventsToTrace(logEventModel, xT);
+            assignMyCaseAttributes(logEventModel.getCaseAttributes(), xT);
+            tracesHistory.put(logEventModel.getCaseID(), xT);
+        } else {
+            XTrace xT = tracesHistory.get(logEventModel.getCaseID());
+            assignEventsToTrace(logEventModel, xT);
+            assignMyCaseAttributes(logEventModel.getCaseAttributes(), xT);
+        }
+    }
+
+    protected void sortAndFeedLog(final TreeMap<String, XTrace> tracesHistory, final XLog xLog) {
+        tracesHistory.forEach(
+            /*
+             * Java 8 (k, v) -> { v.sort(new XEventComparator()); xLog.add(v); }
+             */
+            new java.util.function.BiConsumer<String, XTrace>() {
+                public void accept(String k, XTrace v) {
+                    v.sort(new XEventComparator());
+                    xLog.add(v);
+                }
+            });
```

Comment:
Why isn't the Java 8 syntax being used?

Reply:
This was from a copy/paste while doing a bit of refactoring. I've fixed this up now, as part of addressing sonar cloud code smells. Will address the remaining Java 8 todo comments across the code in a subsequent PR